core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/NamedOperationResolver.java

```diff
@@ -122,42 +95,64 @@ public <T> T onFailure(final T result, final OperationChain<?> opChain, final Co
         return result;
     }
 
-    private void resolveNamedOperations(final Operations<?> operations, final User user) {
-        final List<Operation> updatedOperations = new ArrayList<>();
-
-        operations.getOperations().forEach(operation -> {
-            if (operation instanceof NamedOperation) {
-                updatedOperations.addAll(resolveNamedOperation((NamedOperation<?, ?>) operation, user));
-            } else {
-                if (operation instanceof Operations) {
-                    resolveNamedOperations(((Operations<?>) operation), user);
-                }
-                updatedOperations.add(operation);
+    /**
+     * Resolves any named operations from the cache. What is meant
+     * by 'resolved' is turning the named operations into their respective
+     * {@link OperationChain}s. Ensures any supplied {@link NamedOperation}s
+     * actually exist in the cache and contain their correct {@link Operation}s.
+     * Will also run recursively to a given depth limit to ensure any nested
+     * {@link NamedOperation}s are also resolved from the cache.
+     *
+     * @param operation {@link NamedOperation} or {@link OperationChain} to act on.
+     * @param user      User for the cache access.
+     * @param depth     Current recursive depth, will use limit set in class to
```

Comment:
inclusive or exclusive of the number, does zero resolve nothing?

Reply:
Not quite sure what you mean, the limit is set on a class level this is just the current depth of recursion of the method.

Reply:
Foolish me, I miss read "depth" and "depthLimit"